packages/rancher-monitoring/generated-changes/patch/templates/alertmanager/secret.yaml.patch

```diff
@@ -1,164 +1,28 @@
 --- charts-original/templates/alertmanager/secret.yaml
 +++ charts/templates/alertmanager/secret.yaml
-@@ -1,11 +1,19 @@
+@@ -1,11 +1,17 @@
  {{- if and (.Values.alertmanager.enabled) (not .Values.alertmanager.alertmanagerSpec.useExistingSecret) }}
-+{{- if .Release.IsInstall }}
++{{/* This file is applied when the operation is helm install and the target secret does not exist. */}}
 +{{- $secretName := (printf "alertmanager-%s-alertmanager" (include "kube-prometheus-stack.fullname" .)) }}
-+{{- if (lookup "v1" "Secret" (include "kube-prometheus-stack.namespace" .) $secretName) }}
-+{{- required (printf "Cannot overwrite existing secret %s in namespace %s." $secretName (include "kube-prometheus-stack.namespace" .)) "" }}
-+{{- end }}{{- end }}
++{{- if (not (lookup "v1" "Secret" (include "kube-prometheus-stack.namespace" .) $secretName)) }}
  apiVersion: v1
  kind: Secret
  metadata:
 -  name: alertmanager-{{ template "kube-prometheus-stack.fullname" . }}-alertmanager
-+  name: alertmanager-{{ template "kube-prometheus-stack.fullname" . }}-pre-install
++  name: {{ $secretName }}
    namespace: {{ template "kube-prometheus-stack.namespace" . }}
 -{{- if .Values.alertmanager.secret.annotations }}
    annotations:
-+    "helm.sh/hook": pre-install
-+    "helm.sh/hook-delete-policy": hook-succeeded, hook-failed
++    "helm.sh/hook": pre-install, pre-upgrade
 +    "helm.sh/hook-weight": "3"
++    "helm.sh/resource-policy": keep
```

Comment:
Why do we want to keep the secret now?
If it is ok to keep it, why we deleted it previously?

Reply:
We didn't delete it previously. Before, we kept it by default but provided a way to run a job on uninstall to delete it.

However, I removed that cleanup job because 1) we never exposed it on the UI anyways and I don't think it was specifically requested and 2) it required adding a dependency on another image, which doesn't seem to be necessary when all you would need to do is manually clean up the secret with a single kubectl command.